CoreFeatures/Database-postgresql/PostgresSchema/src/main/java/info/smart_tools/smartactors/database_postgresql/postgres_schema/indexes/IndexDroppers.java

```diff
@@ -0,0 +1,124 @@
+import info.smart_tools.smartactors.ioc.ikey.IKey;
+import info.smart_tools.smartactors.ioc.ioc.IOC;
+import info.smart_tools.smartactors.ioc.key_tools.Keys;
+
+import java.io.IOException;
+import java.io.Writer;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * A set of classes to write CREATE INDEX SQL statements.
+ */
+public class IndexDroppers {
+
+    /**
+     * Set of index creation functions.
+     */
+    private static final Map<String, IndexWriterResolver> INDEX_WRITERS = new HashMap<String, IndexWriterResolver>() {{
+        put("ordered", (indexDefinition) -> IndexDroppers::writeOrderedIndex);
+        put("datetime", (indexDefinition) -> IndexDroppers::writeDatetimeIndex);
+        put("tags", (indexDefinition) -> IndexDroppers::writeTagsIndex);
+        put("fulltext", FulltextIndexWriter::resolveDrop);
+    }};
+
+    /**
+     * Private constructor to avoid instantiation.
+     */
+    private IndexDroppers() {
+    }
+
+    /**
+     * Writes DROP INDEX statements to the SQL statement body.
+     * @param body where to write SQL
+     * @param collection name of the collection
+     * @param options document describing dropping indexes
+     * @throws Exception when something goes wrong
+     */
+    public static void writeDropIndexes(Writer body, CollectionName collection, IObject options) throws Exception {
+        try {
+            if (options == null) {
+                // no indexes definition, ignoring
+                return;
+            }
+            for (String indexType : INDEX_WRITERS.keySet()) {
+                writeDropIndex(indexType, body, collection, options);
+            }
+        } catch (ReadValueException e) {
+            // no indexes definition, ignoring
+        }
+    }
+
+    private static void writeDropIndex(final String indexType, final Writer body, final CollectionName collection, IObject options)
+            throws Exception {
+        IKey fieldNameKey = Keys.getKeyByName("info.smart_tools.smartactors.iobject.ifield_name.IFieldName");
+        Object indexFields = null;
+        try {
+            IFieldName indexDefinitionField = IOC.resolve(fieldNameKey, indexType);
+            indexFields = options.getValue(indexDefinitionField);
+        } catch (ReadValueException e) {
+            // ignoring absence of this index type definition
+            return;
+        }
+
+        List<FieldPath> fieldPaths = new ArrayList<>();
+        if (indexFields == null) {
+            // ignoring absence of this index type definition
+            return;
+        } else if (indexFields instanceof String) {
+            fieldPaths.add(PostgresFieldPath.fromString((String) indexFields));
+        } else if (indexFields instanceof List) {
+            for (Object fieldName : (List) indexFields) {
+                fieldPaths.add(PostgresFieldPath.fromString((String) fieldName));
```

Comment:
!!! May raise ClassCastException if the list items are not Strings.

Reply:
done